src/libs/blueprint/conduit_blueprint_mesh.cpp

```diff
@@ -307,6 +311,98 @@ mesh::verify(const Node &n,
 }
 
 
+//-------------------------------------------------------------------------
+bool mesh::verify_multi_domain(const Node &n,
+                               Node &info)
+{
+    info.reset();
+    bool res = true;
+
+    const std::string proto_name = "mesh";
+
+    if(!n.dtype().is_object() && !n.dtype().is_list())
+    {
+        log_error(info,proto_name,"not an object or a list");
+        res = false;
+    }
+    else
+    {
+        NodeConstIterator itr = n.children();
+        while(itr.has_next())
+        {
+            const Node &chld = itr.next();
+            std::string chld_name;
+            {
+                std::ostringstream oss;
+                if(n.dtype().is_object())
+                {
+                    oss << itr.name();
+                }
+                else
+                {
+                    oss << itr.index();
+                }
+                chld_name = oss.str();
+            }
+
+            if(!mesh::verify_single_domain(chld, info[chld_name]))
+            {
+                log_error(info,proto_name,
+                          "child " + chld_name + " is not a valid mesh");
+                res = false;
+            }
+        }
+
+        log_info(info,proto_name,"is a multi domain mesh");
+    }
+
+    return res;
+}
+
+
+//-----------------------------------------------------------------------------
+bool
+mesh::verify(const Node &n,
+             Node &info)
+{
+    info.reset();
+    bool res = true;
+
+    if(mesh::verify_multi_domain(n, info))
+    {
+        res = true;
+    }
+    else
+    {
+        info.reset();
+        res = mesh::verify_single_domain(n, info);
+    }
+
+    return res;
+}
+
+
+//-------------------------------------------------------------------------
+bool mesh::is_multi_domain(const conduit::Node &n)
+{
+    Node info;
+    return mesh::verify_multi_domain(n, info);
+}
+
+
+//-------------------------------------------------------------------------
+bool mesh::to_multi_domain(const conduit::Node &n,
+                           conduit::Node &dest)
+{
+    dest.reset();
+
```

Comment:
one more thing: the transforms should work for all cases where mesh verify is true. 

For this, we need to also handle the case where the input node is already a multi domain mesh.